### PR TITLE
CAMEL-12462

### PR DESCRIPTION
**Pull request by davsclaus**
See ticket: https://issues.apache.org/jira/browse/CAMEL-12462

Anyone is welcome to take a peak and provide feedback. This PR can be squashed and merged when its done.

We can do this kind of optimisation for other components as well, for example JMS with the CamelJmsDestination header for the JMS queue/topic. However JMS uses a connection pool so its fairly optimised as well, but it would still use a new producer per different queue/topic.

### CODE REVIEW

**Comment by fharms** [comment]
Nice. I really like this feature. 

How difficult would it be to also make camel-jetty support this optimization ?

**Comment by davsclaus** [comment]
The camel-jetty producer is deprecated (not the jetty consumer) - so consider switching to another component for http. But I do think its possible to add support for it, let me take a look.

**Comment by fharms** [comment]
Ohh I missed that. No need for adding support for it then, I just switch to another http component.

**Comment by davsclaus** [comment]
Okay added support for jetty as well, as it was easy. Updated the toD docs a bit more about cache size is not your solution, and added a solution with static `to`.

File: components/camel-http/src/test/java/org/apache/camel/component/http/HttpSendDynamicAwareTest.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        return new RouteBuilder() {
+            @Override
+            public void configure() throws Exception {
+                from("direct:moes")
+                    .toD("http://localhost:" + PORT + "/moes?throwExceptionOnFailure=false&drink=${header.drink}");
+
+                from("direct:joes")
+                    .toD("http://localhost:" + PORT + "/joes?throwExceptionOnFailure=false&drink=${header.drink}");
+
+                from("direct:vokda")
```
[the remark this answers is not in the document]
Comment (davsclaus) [reply components/camel-http/src/test/java/org/apache/camel/component/http/HttpSendDynamicAwareTest.java]: Ah sorry lets get that fixed

**Pull request merged**